src/main/java/com/uid2/admin/vertx/service/PrivateSiteDataRefreshService.java

```diff
@@ -0,0 +1,40 @@
+package com.uid2.admin.vertx.service;
+
+import com.uid2.admin.job.JobDispatcher;
+import com.uid2.admin.job.jobsync.OverallSyncJob;
+import com.uid2.admin.vertx.WriteLock;
+import com.uid2.shared.auth.Role;
+import com.uid2.shared.middleware.AuthMiddleware;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class PrivateSiteDataRefreshService implements IService {
+    private final AuthMiddleware auth;
+    private final WriteLock writeLock;
+
+    public PrivateSiteDataRefreshService(AuthMiddleware auth,
+                                            WriteLock writeLock) {
+        this.auth = auth;
+        this.writeLock = writeLock;
+    }
+
+    @Override
+    public void setupRoutes(Router router) {
+        router.post("/api/private_sites/refresh").blockingHandler(auth.handle((ctx) -> {
```

Comment:
Generally it's a convention to use dashes rather than underscores in URLs.

Reply:
Fixed with dash